### PR TITLE
[NCL-2140] Added detailed info about version difference in alignment …

### PR DESCRIPTION
**Pull request by stanlyDoge**
…reports

### CODE REVIEW

File: reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java [hunk 1]
```diff
@@ -355,7 +355,14 @@ private void updateArtifact(ArtifactReport ar, Set<WhiteArtifact> allArtifacts)
                 } else {
                     different = Collections.emptySet();
                 }
+
                 differentVersion.put(gav, different);
+                for (Iterator<ProductArtifact> iterator = different.iterator(); iterator.hasNext();) {
```
Comment (jsenko) [inline reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java]: Why not use for(ProductArtifact next: different) ?

File: reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java [hunk 2]
```diff
@@ -355,7 +355,14 @@ private void updateArtifact(ArtifactReport ar, Set<WhiteArtifact> allArtifacts)
...
+                    ProductArtifact next = iterator.next();
+                    String type = VersionComparator.difference(gav.getVersion(),
+                            next.getArtifact().getVersion()).toString();
+                    next.setDifferenceType(type);
+                }
```
Comment (jsenko) [inline reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java]: The method seems too big, could you move the code to a new prived method and add some javadoc to it?

**Comment by jsenko** [comment]
Looks good, fix the issues if you want and merge it

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

File: testsuite/src/test/rest/v-1.0/expectedResponse/products/diff.json [hunk 3]
```diff
@@ -28,7 +28,8 @@
             "groupId": "com.example",
             "artifactId": "changed",
             "leftVersion": "1.0.0",
-            "rightVersion": "1.2.0"
+            "rightVersion": "1.2.0",
+            "differenceType": "MINOR";
```
Comment (janinko) [inline testsuite/src/test/rest/v-1.0/expectedResponse/products/diff.json]: Why's there semicolon?

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Pull request merged**